src/replicate/replicate.js

```diff
@@ -101,7 +101,26 @@ function replicate(src, target, opts, returnValue, result) {
     });
   }
 
+  function findLatestSeq() {
+    var lastId = currentBatch.docs[currentBatch.docs.length - 1]._id;
+    for (var i = 0; i < currentBatch.changes.length; i++) {
+      var change = currentBatch.changes[i];
+      if (change.id === lastId) {
+        return change.seq;
+      }
+    }
+  }
+
   function finishBatch() {
+    if (Object.keys(currentBatch.diffs).length) {
+      if (!currentBatch.docs.length) {
+        // changes expected by none written - don't update checkpoint
+        return;
```

Comment:
You can't just return here, replication needs to be either [completed](https://github.com/pouchdb/pouchdb/pull/5033/files#diff-1b1f318b599e364220cdb8ac006c238bR135), [restarted](https://github.com/pouchdb/pouchdb/pull/5033/files#diff-1b1f318b599e364220cdb8ac006c238bR139) or [aborted](https://github.com/pouchdb/pouchdb/pull/5033/files#diff-1b1f318b599e364220cdb8ac006c238bR142). This might be the cause for [this reliably failing test](https://travis-ci.org/pouchdb/pouchdb/jobs/121032787#L3376).

I'd suggest to leave it up to the checkpointer to skip writing the checkpoint if nothing changed; this is out of the scope of this issue anyway imo.